staging/src/k8s.io/kubectl/pkg/polymorphichelpers/rollback.go

```diff
@@ -124,7 +124,7 @@ func (r *DeploymentRollbacker) Rollback(obj runtime.Object, updatedAnnotations m
 		return printTemplate(&rsForRevision.Spec.Template)
 	}
 	if deployment.Spec.Paused {
-		return "", fmt.Errorf("you cannot rollback a paused deployment; resume it first with 'kubectl rollout resume deployment/%s' and try again", name)
+		return "", fmt.Errorf("you cannot rollback a paused deployment; resume it first with 'kubectl rollout resume deployment/%s -n %s' and try again", name, namespace)
```

Comment:
Are you sure that in case of default namespace the error will be "-n default" and not "-n "?

Reply:
This is a good point.  The namespace variable here is coming from the object's `metadata.namespace`, and reading [this comment](https://github.com/kubernetes/kubernetes/blob/a9593d634c6a053848413e600dadbf974627515f/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L136-L137) makes me THINK it should be OK, but it doesn't clear things up 100%:

> An empty namespace is equivalent to the "default" namespace, but "default" is the canonical representation

This implies, to me, that empty will be change to "default".  But on the other hand it implies that it COULD be empty and still be valid.  Am I reading this correctly?  It sounds like we might want to check if namespace is empty, and use default if so, just to be safe.

Also, we can't just omit the -n in this case, even though for many people "default" is their default namespace, you can configure a kubectl context to use a different namespace, so it would get complicated to try to figure out whether "default" is their default namespace in their current kubectl context.

Reply:
There is part of me that thinks maybe we should not be telling people exactly which command to run. Maybe we should be just saying:
```
...resume it first with 'kubectl rollout resume`
```

and not try to build up the entire command for them.

Is kubectl trying to do too much here?

For example, we don't know if they are supposed to be using `--context=production` or something like that.  Telling them to run a specific complete command seems like it could be a somewhat dangerous thing.

I think there is a difference between this and help text examples, because people understand that they need to adapt help text examples to their use case, but when we are including their deployment name and namespace in the completely runnable command, it implies that we are telling them that it is correct to run that specific command.

Am I making sense?  Thoughts about this?

Reply:
I tried and we got -n default.This can be run correctly, but as @brianpursley mentioned, it seems like a little strange.

```
[root@lzx-22:/opt/kubernetes/1.4.5]$ kubectl  rollout undo deployment kubia
error: you cannot rollback a paused deployment; resume it first with 'kubectl rollout resume deployment/kubia -n default' and try again
```
> Are you sure that in case of default namespace the error will be "-n default" and not "-n "?

Reply:
@brianpursley, I agree with you. If we cannot give the correct command, i think it's better to just use `kubectl rollout resume`.

Reply:
> Telling them to run a specific complete command seems like it could be a somewhat dangerous thing.

👍  I also agree with you, Brian. As you're saying, it's not helpful for it to _look_ runnable and complete while potentially not being *exactly* what they need to run under the circumstances.